File: app/src/main/java/com/xingen/camera/PictureActivity.java
package com.xingen.camera;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.util.Log;
import android.widget.ImageView;

import com.xingen.camera.base.BaseActivity;
import com.xingen.camera.glide.GlideApp;

/**
 * Created by ${xingen} on 2017/10/27.
 * <p>
 * 查看拍照图片
 */

public class PictureActivity extends BaseActivity {
    public static final String TAG = PictureActivity.class.getSimpleName();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.i(TAG,TAG+"onCreate");
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_picture;
    }

    @Override
    protected void initView(Bundle savedInstanceState) {
        Bundle bundle = getIntent().getExtras();
        if (bundle != null & bundle.containsKey(TAG)) {
            String url = bundle.getString(TAG);
            ImageView imageView = (ImageView) findViewById(R.id.picture_show_iv);
            GlideApp.with(this).asBitmap().load(url).into(imageView);
        }
    }
    public static void openActivity(Context context, String url) {
        Bundle bundle = new Bundle();
        bundle.putString(PictureActivity.TAG, url);
        Intent intent = new Intent(context, PictureActivity.class);
        intent.putExtras(bundle);
        context.startActivity(intent);
    }
}

File: app/src/main/java/com/xingen/camera/PictureActivity.kt
package com.xingen.camera

import android.content.Context
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.util.Log
import android.widget.ImageView
import com.bumptech.glide.Glide
import com.xingen.camera.base.BaseActivity


/**
 * Created by ${xingen} on 2017/10/27.
 *
 *
 * 查看拍照图片
 */
class PictureActivity : BaseActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.i(TAG, TAG + "onCreate")
    }

    override fun getLayoutId(): Int {
        return R.layout.activity_picture
    }

    override fun initView(savedInstanceState: Bundle?) {
        val bundle = intent.extras
        if ((bundle != null) and bundle!!.containsKey(TAG)) {
            val url = bundle.getParcelable<Uri>(TAG)
            val imageView = findViewById<ImageView>(R.id.picture_show_iv)
            Glide.with(this).asBitmap().load(url).into(imageView)
        }
    }

    companion object {
        val TAG: String = PictureActivity::class.java.getSimpleName()

        fun openActivity(context: Context, uri: Uri ) {
            val bundle = Bundle()
            bundle.putParcelable(TAG, uri)
            val intent = Intent(context, PictureActivity::class.java)
            intent.putExtras(bundle)
            context.startActivity(intent)
        }
    }
}
